src/segment/timelineTimeParser.js

```diff
@@ -1,31 +1,67 @@
-import { range } from '../utils/list';
+/**
+ * Calculates the R (repetition) value for a live stream (for the final segment
+ * in a manifest where the r value is negative 1)
+ *
+ * @param {Object} attributes
+ *        Object containing all inherited attributes from parent elements with attribute
+ *        names as keys
+ * @param {number} time
+ *        current time (typically the total time up until the final segment)
+ * @param {number} duration
+ *        duration property for the given <S />
+ *
+ * @return {number}
+ *        R value to reach the end of the given period
+ */
+const getLiveRValue = (attributes, time, duration) => {
+  const {
+    NOW,
+    clientOffset,
+    availabilityStartTime,
+    timescale = 1,
+    start = 0,
+    minimumUpdatePeriod = 0
+  } = attributes;
+  const now = (NOW + clientOffset) / 1000;
+  const periodStartWC = availabilityStartTime + start;
+  const periodEndWC = now + minimumUpdatePeriod;
+  const periodDuration = periodEndWC - periodStartWC;
+  const segmentDuration = duration / timescale;
+
+  return Math.ceil((periodDuration - time) / segmentDuration);
```

Comment:
I think this should be `Math.ceil(((periodDuration * timescale) - time) / duration);` or `Math.ceil((periodDuration - (time / timescale)) / segmentDuration);`

`periodDuration` and `segmentDuration` are in seconds, while `time` and `duration` are in `timescale` units, so we need to make sure everything is in the same unit before determining the count